python/ray/_private/gcs_utils.py

```diff
@@ -182,13 +196,227 @@ def internal_kv_keys(self, prefix: bytes) -> List[bytes]:
 
     @staticmethod
     def create_from_redis(redis_cli):
-        gcs_address = redis_cli.get("GcsServerAddress")
-        if gcs_address is None:
-            raise RuntimeError("Failed to look up gcs address through redis")
-        return GcsClient(gcs_address.decode())
+        return GcsClient(get_gcs_address_from_redis(redis_cli))
 
     @staticmethod
     def connect_to_gcs_by_redis_address(redis_address, redis_password):
         from ray._private.services import create_redis_client
         return GcsClient.create_from_redis(
             create_redis_client(redis_address, redis_password))
+
+
+class GcsPublisher:
+    """Publisher to GCS."""
+
+    def __init__(self, address: str = None, channel: grpc.Channel = None):
+        if address:
+            assert channel is None, \
+                "address and channel cannot both be specified"
+            channel = create_gcs_channel(address)
+        else:
+            assert channel is not None, \
+                "One of address and channel must be specified"
+        self._stub = gcs_service_pb2_grpc.InternalPubSubGcsServiceStub(channel)
```

Comment:
maybe just add it into gcs client? We can have this class as a wrapper for that. I think we probably need to reconnect for gcs restarts and hope they should be handled in a similar way.

Reply:
As suggested by Eric earlier, `GcsPublisher` and `GcsSubscriber` are moved to a different file. I just found out that GRPC reuses the underlying connection if `ChannelArgs` is the same ([stackoverflow](https://stackoverflow.com/a/54134144/16564296)). I hope by using the same GRPC channel options to create channels, we don't need a separate mechanism to dedup channels.